cfe_internal/update/update_policy.cf

```diff
@@ -300,13 +298,106 @@ bundle agent cfe_internal_update_policy_cpv
 bundle agent cfe_internal_update_cmdb
 # @brief Ensure local cache of CMDB data is up to date
 {
+  methods:
+
+    policy_server.enterprise_edition::
+      "cfe_internal_update_cmdb_data_distribution";
+
+      # Don't pull CMDB data on policy_hub self bootstrap because
+      # there will be no cf-serverd listening to serve files yet.
+    enterprise_edition.!(am_policy_hub.bootstrap_mode)::  # ENT-6840
+      "cfe_internal_update_cmdb_data_consumption";
+}
+
+bundle agent cfe_internal_update_cmdb_data_distribution
+# @brief Ensure data is ready for agents to download
+{
+  vars:
+    policy_server.enterprise_edition::
+
+      # The API response for host specific data from cmdb tells us the timestamp of the last data change
+      # We store this timestamp and use it for the next request.
+      "_cmdb_next_request_state_file"
+        string => "$(sys.statedir)/cmdb_next_request_from.dat";
+
+      # If we have the timestamp from a previous response we use it, else we start from 0
+      "_cmdb_previous_next_request_from"
+        string => readfile( $(_cmdb_next_request_state_file), inf ),
+        if => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      "_cmdb_previous_next_request_from"
+        string => "0",
+        unless => regline( "^\d+$", $(_cmdb_next_request_state_file) );
```

Comment:
This would be easier to read for me if it used `ifelse()`.

Reply:
Yeah, I tried to do that, but when the state file did not exist, I got errors thrown from readfile.

Perhaps I should file an issue for that.

Reply:
Filed issue. ENT-7202